security/class-private-sites.php

```diff
@@ -71,6 +70,54 @@ public function disable_core_feeds() {
 		add_action( 'do_feed_atom', array( $this, 'action_do_feed' ), -1 );
 	}
 
+	/**
+	 * Force the blog_public option to be -1 and disable UI
+	 */
+	public function force_blog_public_option() {
+		add_filter( 'blog_privacy_selector', '__return_true' );
+		add_filter( 'option_blog_public', array( $this, 'filter_restrict_blog_public' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'disable_blog_public_ui' ) );
+	}
+
+	/**
+	 * Disable checkbox/radio UI in Reading Settings
+	 */
+	public function disable_blog_public_ui() {
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		$screen = get_current_screen();
+		if ( 'options-reading' !== $screen->base ) {
+			return;
+		}
+
+		wp_register_script( 'vip-disable-blog-public-option-ui', false, array(), '0.1', true );
+		wp_enqueue_script( 'vip-disable-blog-public-option-ui' );
+		$js_code       = <<<JS
+		document.addEventListener("DOMContentLoaded", function() {
+			function updateProperty(selector, property, value) {
+				const element = document.querySelector(selector);
+				if (element) {
+					element[property] = value;
+				}
+			}
+
+			var checkbox = 'tr.option-site-visibility input#blog_public[type="checkbox"]';
+			if (document.querySelector(checkbox)) {
+				updateProperty(checkbox, 'disabled', true);
+			} else {
+				updateProperty('tr.option-site-visibility input#blog-public[type="radio"]', 'disabled', true);
+				updateProperty('tr.option-site-visibility input#blog-norobots[type="radio"]', 'disabled', true);
+			}
+			updateProperty('tr.option-site-visibility p.description', 'textContent', '%s');
+		});
```

Comment:
```suggestion
		function onContentLoaded(callback) {
			if (document.readyState !== 'loading') {
				callback();
			} else {
				document.addEventListener('DOMContentLoaded', callback);
			}
		}

		onContentLoaded(function() {
			function updateProperty(selector, property, value) {
				const element = document.querySelector(selector);
				if (element) {
					element[property] = value;
				}
			}
			var checkbox = 'tr.option-site-visibility input#blog_public[type="checkbox"]';
			if (document.querySelector(checkbox)) {
				updateProperty(checkbox, 'disabled', true);
			} else {
				updateProperty('tr.option-site-visibility input#blog-public[type="radio"]', 'disabled', true);
				updateProperty('tr.option-site-visibility input#blog-norobots[type="radio"]', 'disabled', true);
			}
			updateProperty('tr.option-site-visibility p.description', 'textContent', '%s');
		});
```

Reply:
I don't know if our clients use Cloudflare or any kind of site load optimization tools, but if they do, `DOMContentLoaded` may not fire.

For example, CLoudflare's RocketLoader loads script on `DOMContentLoaded` event. This means that if you set another event listener for `DOMContentLoaded`, it is too late; it won't fire.

The solution here is to use [`document.readyState`](https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState). When its value is `loading`, setting our `DOMContentLoaded` handler is OK. If it is set to something else, this means that the `DOMContentLoaded` (`interactive`) or `load` (`complete`) event has already fired and we need to invoke our handler immediately.